packages/core/helper-plugin/src/components/Table.tsx

```diff
@@ -46,14 +46,14 @@ interface TableContextValue<TRow extends { id: number } = { id: number }> {
 
 const TableContext = React.createContext<TableContextValue | null>(null);
 
-const useTableContext = () => {
+const useTableContext = <TRow extends { id: Entity.ID }>() => {
   const context = React.useContext(TableContext);
 
   if (!context) {
     throw new Error('useTableContext must be used within a TableProvider');
   }
 
-  return context;
+  return context as TableContextValue<TRow>;
```

Comment:
Is this cast the only solution here? Can't you create a context with a type?

Reply:
unfortunately yes, so the hook won't know about what TRow _really_ is hence why it's a generic for the context getter, but then if you pass `TableContextValue<TRow>` to the specific `useContext` hook that doesn't work either because TRow doesn't match up with the default value since TRow is expected to extend past that...